context/app/static/js/components/home/EntityCounts/hooks.js

```diff
@@ -0,0 +1,27 @@
+import { useEffect, useState } from 'react';
+
+import useSearchData from 'js/hooks/useSearchData';
+
+const entityCountsQuery = {
+  size: 0,
+  aggs: { entity_type: { terms: { field: 'entity_type.keyword' } } },
+};
+
+function useEntityCounts() {
+  const [entityCounts, setEntityCountsData] = useState(undefined);
+  const { searchData: elasticsearchData } = useSearchData(entityCountsQuery);
+  useEffect(() => {
+    if (Object.keys(elasticsearchData).length) {
+      const entityCountsObject = elasticsearchData.aggregations.entity_type.buckets.reduce((acc, entity) => {
+        const accCopy = acc;
+        accCopy[entity.key] = entity.doc_count;
+        return accCopy;
+      }, {});
```

Comment:
Slight preference for named function that explains what's happening.

Reply:
I think this is similar to a python comprehension in its complexity, but I can pull it to a function.

Reply:
It is older code that could be simplified as well.